tests/instrumentation/transactions_store_tests.py

```diff
@@ -131,6 +131,34 @@ def test_get_transaction_clear():
     assert get_transaction() is None
 
 
+def test_should_collect_time():
+    requests_store = TransactionsStore(lambda: [], collect_frequency=5)
+    requests_store._last_collect -= 6
+
+    assert requests_store.should_collect()
+
+
+def test_should_not_collect_time():
+    requests_store = TransactionsStore(lambda: [], collect_frequency=5)
+    requests_store._last_collect -= 3
+
+    assert not requests_store.should_collect()
+
+
+def test_should_collect_count():
+    requests_store = TransactionsStore(lambda: [], collect_frequency=5, max_queue_length=5)
+    requests_store._transactions = 6*[1]
+
+    assert requests_store.should_collect()
```

Comment:
could you also set `requests_store._last_collect -= 3` to show that the count overrides the time?